automata/base/automaton.py

```diff
@@ -8,12 +8,24 @@
 import automata.base.config as global_config
 import automata.base.exceptions as exceptions
 from automata.base.utils import freeze_value
+from typing import Any, Mapping, NoReturn, AbstractSet, Tuple, Dict, Generator
 
+AutomatonStateT = Any
+PathT = Mapping[str, Any]
```

Comment:
@eliotwrobson I'm guessing the fact that you've set the type for the path value as `Any` is part of solution to solving the Liskov problem (at least from a type perspective)? I was thinking about this earlier, and that's exactly what I was going to suggest doing (at least for the abstract base classes).

I also see you're getting more specific at the subclass level as well (e.g. with `NFATransitionsT`), which was exactly what I was envisioning. So thanks for that! 😀

I know you also mentioned some sort of wrapper to solve the Liskov problem—curious to hear more about what you had in mind with that.

Reply:
Yes, the `Any` here is to allow the derived classes to override this. The Liskov issue actually wasn't with this particular type, but it came up when annotating some validation function in one of the TM classes or something.